protostar/commands/test/cheatcodes/deploy_cheatcode.py

```diff
@@ -32,7 +37,15 @@ def name(self) -> str:
     def build(self) -> Callable[[Any], Any]:
         return self.deploy_prepared
 
-    def deploy_prepared(self, prepared: PreparedContract):
+    def deploy_prepared(
+        self,
+        prepared: PreparedContract,
+        *args,
+        # pylint: disable=unused-argument
+        config: Optional[CheatcodeNetworkConfig] = None,
```

Comment:
Passing in a dict won't throw an error anyway and changing it here removes protocol type safety

Reply:
@kasperski95 I do not understand your suggestion and it seems like a regression from using TypedDict. Could you please elaborate, why the `Dict` is needed here? Aren't test cheatcodes expecting a NetworkConfig-like object even though they are not using it?

Reply:
In short: Interface Segregation Principle

Test cheatcodes should not crash when config argument is passed, and that's the main reason behind the Protocol. They shouldn't care about Network Configuration.

Reply:
They wouldn't crash anyway because Python typing is merely a hint.

Reply:
I know, I phrased it incorrectly. Types are used here to prevent the developer working on Migrator or Testing Cheatcode from creating incompatible API by accident, because Protostar will crash.

Reply:
Hmm; so why do we have the `config` argument then after all? Because original motivation for it was to maintain shared interface between migrator and tests.

Reply:
An unused argument is the sacrifice that should allow the migration scripts to be testable. If the migration script uses config and the script is run in TestExecEnv the test will crash.